### DISCUSSION TITLE
Passenger time out

### DISCUSSION THREAD

**Discussion by Yelia41** [remix-run/remix#4174] [Q&A]
Hello,

For days I have been desperately trying to get my Remix app online via Cpanel (o2switch) except NodeJS apps are deployed by Passenger and this seems to be a problem.
I've tried everything and I can't.
I always fall into time out.
A priori here is what should be done but I can't find where to do it in the REMIX app.
https://faq.o2switch.fr/hebergement-mutualise/tutoriels-cpanel/app-nodejs#timeout-l-application-ne-se-lance-pas
Can you help me ?
Thanks a lot

Ps: I always managed to deploy App NodeJs without problems, even Strapi...

Ps 2 : I try this but it not working :(
https://github.com/kiliman/remix-ftp-deploy 
Same probleme

**Comment by Yelia41** [comment]
Up please. You can see the time out here : https://preprod.evag.fr/

**Reply by kiliman** [reply]
What adapter are you using? Remix App Server or Express?

How is your app configured on your host? What is the startup? Did you install the node modules?

![06C6553F-CADB-4564-BD00-62FAC0CDC82F](https://user-images.githubusercontent.com/47168/192635843-78a6fa08-9123-415b-819c-a385b3aac580.jpeg)

**Reply by Yelia41** [reply]
Hello,
Thank you very much for your answer, so :
I use Remix App Server. I followed this instructions, the startup is build/index.js. Node modules are installed it's ok.
Just for information, there are my commands : 
`  "scripts": {
    "build": "npm run build:css && remix build",
    "build:css": "tailwindcss -m -i ./styles/app.css -o app/styles/app.css",
    "dev": "concurrently \"npm run dev:css\" \"remix dev\"",
    "dev:css": "tailwindcss -w -i ./styles/app.css -o app/styles/app.css",
    "start": "remix-serve build"
  },`

And in local I launch "npm run dev" and it's ok.

Thank you !!!

**Reply by Yelia41** [reply]
Up please ! Thanks

**Reply by Yelia41** [reply]
Help please !!!

**Reply by kiliman** [reply]
So are you saying it's no longer working? Did it ever work?

**Reply by Yelia41** [reply]
It works very well locally and on a VPS server but impossible to make it work on my o2switch hosting and I need it :(
I manage to launch other NodeJS applications perfectly well on it

**Reply by kiliman** [reply]
Have you contacted their support?

**Reply by Yelia41** [reply]
o2switch support ?
Yes, they say there is no problem on their end... I don't know what to do anymore

**Reply by kiliman** [reply]
Unfortunately, since I'm not using that host, I don't know what else to tell you, other than consider a different host?

**Reply by Yelia41** [reply]
CPanel is famous, try to install it on your servers or locally to test

**Comment by schizek** [comment]
@Yelia41 did you ever figure out your issue?

I'm having the exact same problem although I'm still unclear how the files need to be setup within cPanel.  I'm running cPanel on DigitalOcean hosting using "Application Manager" (Phusion Passenger).  Application Manager within cPanel does not provide the ability to set the name of the starting file and requires the file to be named `app.js` and placed in the root directory.

These are the steps I followed so far although I'm unclear if this is the correct procedure (I'm assuming it's not since it won't work)...

1. In cPanel setup Application Manager to point to a directory on my server where my files will be uploaded.
2. On local, ran `npm run build`
3. Uploaded the following files from my local to my server...
    * `/package.json`
    * `/package-lock.json`
    * `/public/`
    * `/build/index.js` (renamed as `/app.js` and moved to the root directory)
4. On server, ran `npm install`
5. In the browser, tried to access the domain I configured within Application Manager
6. At this point the browser tries loading for a while and eventually throws the generic Passenger error "We're sorry, but something went wrong. The issue has been logged for investigation. Please try again later."
7. While I attempted step 5 above, I was watching my logs with the following command on my server... `tail -f /var/log/apache2/error_log`
8. Once the browser timed out, the following error appeared in the log...
    ```
    [ E 2022-12-16 12:44:02.0969 14054/Tr age/Cor/App/Implementation.cpp:221 ]: Could not spawn process for application /home/<myuser>/<mydocroot>: A timeout occurred while spawning an application process.
      Error ID: b139c1b6
      Error details saved to: /var/run/ea-ruby27-passenger/passenger-error-fU496V.html
    
    [ E 2022-12-16 12:44:02.1041 14054/T5 age/Cor/Con/CheckoutSession.cpp:283 ]: [Client 1-4] Cannot checkout session because a spawning error occurred. The identifier of the error is b139c1b6. Please see earlier logs for details about the error.
    ```
9. In the `/var/run/ea-ruby27-passenger/passenger-error-fU496V.html` file mentioned above I'm getting this...
    <img width="816" alt="Screenshot 2022-12-16 at 6 52 11 AM" src="https://user-images.githubusercontent.com/5023509/208102388-a2deca1f-485c-47e6-acf6-144a8dd50293.png">

I've taken a look at [the solution](https://github.com/kiliman/remix-ftp-deploy) @kiliman provided, but I'm not particularly clear on how to implement it.  I'm not currently interested in automating deployment, though... I'm just trying to get it running to test things out and a proof of concept so I can understand the process of what is needed.  Although, I'm sure that will be a welcomed addition once I get closer to actually needing to deploy more regularly.

If anyone has any suggestions it would be much appreciated.  Thanks!

**Reply by Yelia41** [reply]
Hello,
No, I still haven't found a solution I'm desperate...
